src/plugins/intel_gpu/src/graph/include/pass_manager.h

```diff
@@ -323,15 +323,20 @@ class memory_dependency_pass : public base_pass {
             return;
         }
 
+        auto mem_deps = node->get_memory_dependencies();
+        if (mem_deps.find(static_cast<uint32_t>(dep->get_unique_id())) != mem_deps.end()) {
+            return;
```

Comment:
could you add a comment here?
`// If this dependency is already there, exit early`

Reply:
Done.